src/errors/errors.ts

```diff
@@ -1,153 +1,209 @@
- * either delete the existing cache and make a new one, or change the name of the cache you are trying to create to
- * one that doesn't already exist
+ * Error when an operation with Cache Service was cancelled
  */
-export class AlreadyExistsError extends CacheServiceError {
-  constructor(message: string) {
-    super(message);
-    this.name = 'AlreadyExistsError';
-  }
+export class CancelledError extends SdkError {
+  _errorCode = MomentoErrorCode.CANCELLED_ERROR;
+  _messageWrapper =
+    'The request was cancelled by the server; please contact us at support@momentohq.com';
 }
 
 /**
- * Error that occurs when trying to get a cache that doesn't exist. To resolve, make sure that the cache you are trying
- * to get exists. If it doesn't create it first and then try again
+ * Error raised when system in not in a state required for the operation's success
  */
-export class NotFoundError extends CacheServiceError {
-  constructor(message: string) {
-    super(message);
-    this.name = 'NotFoundError';
-  }
+export class FailedPreconditionError extends SdkError {
+  _errorCode = MomentoErrorCode.FAILED_PRECONDITION_ERROR;
+  _messageWrapper =
+    "System is not in a state required for the operation's execution";
 }
 
 /**
  * Cache Service encountered an unexpected exception while trying to fulfill the request
  */
-export class InternalServerError extends CacheServiceError {
-  constructor(message: string, stack: string) {
-    super(message);
-    this.name = 'InternalServerError';
-    this.stack = stack;
-  }
+export class InternalServerError extends SdkError {
+  _errorCode = MomentoErrorCode.INTERNAL_SERVER_ERROR;
+  _messageWrapper =
+    'An unexpected error occurred while trying to fulfill the request; please contact us at support@momentohq.com';
 }
 
 /**
- * Insufficient permissions to perform an operation on Cache Service
+ * Represents errors thrown when invalid parameters are passed to the Momento Cache
  */
-export class PermissionError extends CacheServiceError {
-  constructor(message: string) {
-    super(message);
-    this.name = 'PermissionError';
-  }
+export class InvalidArgumentError extends SdkError {
+  _errorCode = MomentoErrorCode.INVALID_ARGUMENT_ERROR;
+  _messageWrapper = 'Invalid argument passed to Momento client';
 }
 
 /**
- * Error when authentication with Cache Service fails
+ * Error when calls are throttled due to request limit rate
  */
-export class AuthenticationError extends CacheServiceError {
-  constructor(message: string) {
-    super(message);
-    this.name = 'AuthenticationError';
-  }
+export class LimitExceededError extends SdkError {
+  _errorCode = MomentoErrorCode.LIMIT_EXCEEDED_ERROR;
+  _messageWrapper =
+    'Request rate exceeded the limits for this account.  To resolve this error, reduce your request rate, or contact us at support@momentohq.com to request a limit increase';
```

Comment:
Limit errors can be for bandwidth and object size as well. Maybe reword this message a bit to be more generic vs just 'rate'?

Reply:
@eaddingtonwhite This error message is copied from the [Standards and Practices Error Handling doc](https://github.com/momentohq/standards-and-practices/blob/main/docs/client-specifications/error-handling.md) so it should be updated there as well.